lib/ansible/utils/unsafe_proxy.py

```diff
@@ -110,5 +115,8 @@ def wrap_var(v):
     elif isinstance(v, Set):
         v = _wrap_set(v)
     elif v is not None and not isinstance(v, AnsibleUnsafe):
-        v = UnsafeProxy(v)
+        if isinstance(v, binary_type):
```

Comment:
first thing wrap_var should do is test if the object is already 'unsafe' and return it